pkg/ddl/index.go

```diff
@@ -1670,6 +1673,21 @@ type addIndexIngestWorker struct {
 	jobID    int64
 }
 
+func getLocalWriterConfig(indexCnt, writerCnt int) *backend.LocalWriterConfig {
+	writerCfg := &backend.LocalWriterConfig{}
+	// avoid unit test panic
+	memRoot := ingest.LitMemRoot
+	if memRoot == nil {
+		return writerCfg
+	}
+
+	availMem := memRoot.MaxMemoryQuota() - memRoot.CurrentUsage()
+	memLimitPerWriter := availMem / int64(indexCnt) / int64(writerCnt)
```

Comment:
What will happen if `availMem` is very small?

Reply:
it will trigger KV in memory flushing to SST files very often, and the performance will drop 🤔

Reply:
Do you mean also manager the resource usage of txn add index? I'll add the TODO in next PR https://github.com/pingcap/tidb/issues/53858